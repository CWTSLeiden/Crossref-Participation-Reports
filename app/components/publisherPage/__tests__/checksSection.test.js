import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon'
import ChecksSection from "../components/checksSection"





//Dummy Data
const coverage = {
  'journal-articles': [
    {name: 'ja item 1', percentage: 15, info: 'some tooltip info'},
    {name: 'ja item 2', percentage: 25, info: 'some tooltip info'},
  ],

  'books': [
    {name: 'books item 1', percentage: 15, info: 'some tooltip info'},
    {name: 'books item 2', percentage: 25, info: 'some tooltip info'},
    {name: 'books item 2', percentage: 25, info: 'some tooltip info'},
  ]
}


//Mocks
const sandbox = sinon.sandbox.create()
let setState = sandbox.spy(ChecksSection.prototype, 'setState')



//Render and begin tests
const shallowComponent = shallow(<ChecksSection coverage={coverage}/>)



test('shallow snapshot', () => {
  expect(toJson(shallowComponent)).toMatchSnapshot()
})


describe('setOpenTooltip', () => {
  const openTooltip = 'Abstracts'

  test('should call setState with openTooltip and set CheckBox prop', () => {
    sandbox.resetHistory()
    setState.restore()

    const instance = shallowComponent.instance()

    const checkSetStateReturn = sandbox.stub()
    setState = sandbox.stub(instance, 'setState').callsFake( (newState) => {
      if(typeof newState === 'function') {
        const prevState = instance.state
        checkSetStateReturn(newState(prevState))
      } else {
        checkSetStateReturn(newState)
      }
    })

    instance.setOpenTooltip(openTooltip)
    expect(setState.callCount).toBe(1)
    expect(checkSetStateReturn.calledWithMatch({openTooltip})).toBe(true)
    setState.restore()
    setState = sandbox.spy(instance, 'setState')

    shallowComponent.setState({openTooltip})
    shallowComponent.update()
    expect(shallowComponent.find('CheckBox').first().prop('openTooltip')).toBe(openTooltip)
  })
})


describe('filter is set to new value', () => {
  let newFilter = 'books'

  test('new checks are rendered', () => {
    sandbox.resetHistory()
    const instance = shallowComponent.instance()
    instance.setFilter(newFilter)
    expect(setState.callCount).toBe(1)
    shallowComponent.update()
    const checks = shallowComponent.find('CheckBox')
    expect(checks.length).toBe(coverage[newFilter].length)
    expect(toJson(shallowComponent)).toMatchSnapshot()
  })

  test('titleBar filter type text updated', () => {
    expect(shallowComponent.find('.titleBar').text()).toBe(`Content type: Books`)
  })
})
