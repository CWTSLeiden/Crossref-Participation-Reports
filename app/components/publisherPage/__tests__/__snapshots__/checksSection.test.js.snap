// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter is set to new value new checks are rendered 1`] = `
<div
  className="checksSection"
>
  <div
    className="titleBar"
  >
    Content type: Books
  </div>
  <div
    className="filters"
  >
    <ContentTypeFilter
      currentFilter="books"
      filters={
        Array [
          "journal-articles",
          "books",
          "reports",
        ]
      }
      setFilter={[Function]}
    />
    <div
      className="filter publicationFilter"
    >
      Publication Filter
    </div>
    <div
      className="filter timeFilter"
    >
      Last 12 months
    </div>
  </div>
  <div
    className="checksContainer"
  >
    <div
      className="check"
      key="books item 1"
    >
      <div
        className="title"
      >
        books item 1
      </div>
      <div
        className="barContainer"
      >
        <div
          className="bar"
        >
          <div
            className="barWidth"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            <div
              className="animateBounce"
              style={
                Object {
                  "backgroundColor": "#3EB1CB",
                }
              }
            />
          </div>
        </div>
        <div
          className="percent"
        >
          15
          <span>
            %
          </span>
        </div>
      </div>
    </div>
    <div
      className="check"
      key="books item 2"
    >
      <div
        className="title"
      >
        books item 2
      </div>
      <div
        className="barContainer"
      >
        <div
          className="bar"
        >
          <div
            className="barWidth"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="animateBounce"
              style={
                Object {
                  "backgroundColor": "#3EB1CB",
                }
              }
            />
          </div>
        </div>
        <div
          className="percent"
        >
          25
          <span>
            %
          </span>
        </div>
      </div>
    </div>
    <div
      className="check"
      key="books item 2"
    >
      <div
        className="title"
      >
        books item 2
      </div>
      <div
        className="barContainer"
      >
        <div
          className="bar"
        >
          <div
            className="barWidth"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="animateBounce"
              style={
                Object {
                  "backgroundColor": "#3EB1CB",
                }
              }
            />
          </div>
        </div>
        <div
          className="percent"
        >
          25
          <span>
            %
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`shallow snapshot 1`] = `
<div
  className="checksSection"
>
  <div
    className="titleBar"
  >
    Content type: Journal articles
  </div>
  <div
    className="filters"
  >
    <ContentTypeFilter
      currentFilter="journal-articles"
      filters={
        Array [
          "journal-articles",
          "books",
          "reports",
        ]
      }
      setFilter={[Function]}
    />
    <div
      className="filter publicationFilter"
    >
      Publication Filter
    </div>
    <div
      className="filter timeFilter"
    >
      Last 12 months
    </div>
  </div>
  <div
    className="checksContainer"
  >
    <div
      className="check"
      key="ja item 1"
    >
      <div
        className="title"
      >
        ja item 1
      </div>
      <div
        className="barContainer"
      >
        <div
          className="bar"
        >
          <div
            className="barWidth"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            <div
              className="animateBounce"
              style={
                Object {
                  "backgroundColor": "#3EB1CB",
                }
              }
            />
          </div>
        </div>
        <div
          className="percent"
        >
          15
          <span>
            %
          </span>
        </div>
      </div>
    </div>
    <div
      className="check"
      key="ja item 2"
    >
      <div
        className="title"
      >
        ja item 2
      </div>
      <div
        className="barContainer"
      >
        <div
          className="bar"
        >
          <div
            className="barWidth"
            style={
              Object {
                "width": "25%",
              }
            }
          >
            <div
              className="animateBounce"
              style={
                Object {
                  "backgroundColor": "#3EB1CB",
                }
              }
            />
          </div>
        </div>
        <div
          className="percent"
        >
          25
          <span>
            %
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
